docker/influxdb/setup-influxdb.sh

```diff
@@ -0,0 +1,8 @@
+#! /bin/sh
+echo "Creating $INFLUXDB_IML_DB and $INFLUXDB_STRATAGEM_SCAN_DB databases."
+while ! influx -execute "CREATE DATABASE $INFLUXDB_IML_DB" >/dev/null; do
+    echo "influx not yet ready"
+    sleep 1
+done
+
+influx -execute "CREATE DATABASE $INFLUXDB_STRATAGEM_SCAN_DB"
```

Comment:
Guessing these calls are idempotent if the db already exists?

Reply:
Good question. Scripts inside the docker-entrypoint-initdb.d are only executed on the initial startup. For example, If I clear my stack and deploy a new stack the logs look like this:

```
[root@onyx-25 docker]# docker service logs -f iml_influxdb
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | influxdb init process in progress...
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:38.787943Z lvl=info msg="InfluxDB starting" log_id=0KWwR3Ol000 version=1.7.6 branch=1.7 commit=01c8dd416270f424ab0c40f9291e269ac6921964
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:38.787979Z lvl=info msg="Go runtime" log_id=0KWwR3Ol000 version=go1.11 maxprocs=32
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080144Z lvl=info msg="Using data dir" log_id=0KWwR3Ol000 service=store path=/var/lib/influxdb/data
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080423Z lvl=info msg="Compaction settings" log_id=0KWwR3Ol000 service=store max_concurrent_compactions=16 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080462Z lvl=info msg="Open store (start)" log_id=0KWwR3Ol000 service=store trace_id=0KWwR4Y0000 op_name=tsdb_open op_event=start
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080604Z lvl=info msg="Open store (end)" log_id=0KWwR3Ol000 service=store trace_id=0KWwR4Y0000 op_name=tsdb_open op_event=end op_elapsed=0.146ms
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080669Z lvl=info msg="Opened service" log_id=0KWwR3Ol000 service=subscriber
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080682Z lvl=info msg="Starting monitor service" log_id=0KWwR3Ol000 service=monitor
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080692Z lvl=info msg="Registered diagnostics client" log_id=0KWwR3Ol000 service=monitor name=build
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080702Z lvl=info msg="Registered diagnostics client" log_id=0KWwR3Ol000 service=monitor name=runtime
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080712Z lvl=info msg="Registered diagnostics client" log_id=0KWwR3Ol000 service=monitor name=network
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080726Z lvl=info msg="Registered diagnostics client" log_id=0KWwR3Ol000 service=monitor name=system
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080758Z lvl=info msg="Starting precreation service" log_id=0KWwR3Ol000 service=shard-precreation check_interval=10m advance_period=30m
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080776Z lvl=info msg="Starting snapshot service" log_id=0KWwR3Ol000 service=snapshot
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080790Z lvl=info msg="Starting continuous query service" log_id=0KWwR3Ol000 service=continuous_querier
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080816Z lvl=info msg="Starting HTTP service" log_id=0KWwR3Ol000 service=httpd authentication=false
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080825Z lvl=info msg="opened HTTP access log" log_id=0KWwR3Ol000 service=httpd path=stderr
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080867Z lvl=info msg="Listening on HTTP" log_id=0KWwR3Ol000 service=httpd addr=127.0.0.1:8086 https=false
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080883Z lvl=info msg="Starting retention policy enforcement service" log_id=0KWwR3Ol000 service=retention check_interval=30m
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.080860Z lvl=info msg="Storing statistics" log_id=0KWwR3Ol000 service=monitor db_instance=_internal db_rp=monitor interval=10s
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.081001Z lvl=info msg="Sending usage statistics to usage.influxdata.com" log_id=0KWwR3Ol000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:39.081040Z lvl=info msg="Listening for signals" log_id=0KWwR3Ol000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:40 +0000] "GET /ping HTTP/1.1" 204 0 "-" "InfluxDBShell/1.7.6" 8dc58ac5-3e10-11ea-8001-02420a000408 301
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:40.716589Z lvl=info msg="Executing query" log_id=0KWwR3Ol000 service=query query="SHOW DATABASES"
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:40 +0000] "POST /query?chunked=true&db=&epoch=ns&q=SHOW+DATABASES HTTP/1.1" 200 94 "-" "InfluxDBShell/1.7.6" 8dc5ad86-3e10-11ea-8002-02420a000408 1314
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | /init-influxdb.sh: running /docker-entrypoint-initdb.d/setup-influxdb.sh
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | Creating iml and iml_stratagem_scans databases.
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:41 +0000] "GET /ping HTTP/1.1" 204 0 "-" "InfluxDBShell/1.7.6" 8e55060b-3e10-11ea-8003-02420a000408 43
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:41.656443Z lvl=info msg="Executing query" log_id=0KWwR3Ol000 service=query query="CREATE DATABASE iml"
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:41 +0000] "POST /query?chunked=true&db=&epoch=ns&q=CREATE+DATABASE+iml HTTP/1.1" 200 57 "-" "InfluxDBShell/1.7.6" 8e551a7c-3e10-11ea-8004-02420a000408 41640
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:42 +0000] "GET /ping HTTP/1.1" 204 0 "-" "InfluxDBShell/1.7.6" 8ed6a65b-3e10-11ea-8005-02420a000408 44
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.505784Z lvl=info msg="Executing query" log_id=0KWwR3Ol000 service=query query="CREATE DATABASE iml_stratagem_scans"
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | [httpd] 127.0.0.1 - - [23/Jan/2020:18:45:42 +0000] "POST /query?chunked=true&db=&epoch=ns&q=CREATE+DATABASE+iml_stratagem_scans HTTP/1.1" 200 57 "-" "InfluxDBShell/1.7.6" 8ed6b454-3e10-11ea-8006-02420a000408 60181
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    |
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.567834Z lvl=info msg="Signal received, initializing clean shutdown..." log_id=0KWwR3Ol000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.567873Z lvl=info msg="Waiting for clean shutdown..." log_id=0KWwR3Ol000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568011Z lvl=info msg="Shutting down monitor service" log_id=0KWwR3Ol000 service=monitor
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568055Z lvl=info msg="Listener closed" log_id=0KWwR3Ol000 service=snapshot
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568079Z lvl=info msg="Terminating storage of statistics" log_id=0KWwR3Ol000 service=monitor
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568126Z lvl=info msg="Terminating precreation service" log_id=0KWwR3Ol000 service=shard-precreation
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568150Z lvl=info msg="Terminating continuous query service" log_id=0KWwR3Ol000 service=continuous_querier
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568189Z lvl=info msg="Closing retention policy enforcement service" log_id=0KWwR3Ol000 service=retention
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568308Z lvl=info msg="Closed service" log_id=0KWwR3Ol000 service=subscriber
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:42.568376Z lvl=info msg="Server shutdown completed" log_id=0KWwR3Ol000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.409679Z lvl=info msg="InfluxDB starting" log_id=0KWwRLSG000 version=1.7.6 branch=1.7 commit=01c8dd416270f424ab0c40f9291e269ac6921964
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.409719Z lvl=info msg="Go runtime" log_id=0KWwRLSG000 version=go1.11 maxprocs=32
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.510829Z lvl=info msg="Using data dir" log_id=0KWwRLSG000 service=store path=/var/lib/influxdb/data
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.510910Z lvl=info msg="Compaction settings" log_id=0KWwRLSG000 service=store max_concurrent_compactions=16 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.510924Z lvl=info msg="Open store (start)" log_id=0KWwRLSG000 service=store trace_id=0KWwRLqW000 op_name=tsdb_open op_event=start
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511024Z lvl=info msg="Open store (end)" log_id=0KWwRLSG000 service=store trace_id=0KWwRLqW000 op_name=tsdb_open op_event=end op_elapsed=0.102ms
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511063Z lvl=info msg="Opened service" log_id=0KWwRLSG000 service=subscriber
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511071Z lvl=info msg="Starting monitor service" log_id=0KWwRLSG000 service=monitor
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511091Z lvl=info msg="Registered diagnostics client" log_id=0KWwRLSG000 service=monitor name=build
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511096Z lvl=info msg="Registered diagnostics client" log_id=0KWwRLSG000 service=monitor name=runtime
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511101Z lvl=info msg="Registered diagnostics client" log_id=0KWwRLSG000 service=monitor name=network
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511109Z lvl=info msg="Registered diagnostics client" log_id=0KWwRLSG000 service=monitor name=system
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511134Z lvl=info msg="Starting precreation service" log_id=0KWwRLSG000 service=shard-precreation check_interval=10m advance_period=30m
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511144Z lvl=info msg="Starting snapshot service" log_id=0KWwRLSG000 service=snapshot
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511155Z lvl=info msg="Starting continuous query service" log_id=0KWwRLSG000 service=continuous_querier
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511165Z lvl=info msg="Starting HTTP service" log_id=0KWwRLSG000 service=httpd authentication=false
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511174Z lvl=info msg="opened HTTP access log" log_id=0KWwRLSG000 service=httpd path=stderr
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511256Z lvl=info msg="Listening on HTTP" log_id=0KWwRLSG000 service=httpd addr=10.0.4.8:8086 https=false
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511273Z lvl=info msg="Starting retention policy enforcement service" log_id=0KWwRLSG000 service=retention check_interval=30m
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511219Z lvl=info msg="Storing statistics" log_id=0KWwRLSG000 service=monitor db_instance=_internal db_rp=monitor interval=10s
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511411Z lvl=info msg="Listening for signals" log_id=0KWwRLSG000
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:45:43.511469Z lvl=info msg="Sending usage statistics to usage.influxdata.com" log_id=0KWwRLSG000
```

The important piece to take away from the logs above are:
```
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | /init-influxdb.sh: running /docker-entrypoint-initdb.d/setup-influxdb.sh
iml_influxdb.1.gz40yfcreeos@onyx-25.onyx.whamcloud.com    | Creating iml and iml_stratagem_scans databases.
```

Now, let's say I stop the stack with `docker stack rm iml` and then deploy the stack again. This time, we should not see the `Creating iml and iml_stratagem_scans databases.` messages. 

```
[root@onyx-25 docker]# docker service logs iml_influxdb
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.891570Z lvl=info msg="InfluxDB starting" log_id=0KWwqBfl000 version=1.7.6 branch=1.7 commit=01c8dd416270f424ab0c40f9291e269ac6921964
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.891618Z lvl=info msg="Go runtime" log_id=0KWwqBfl000 version=go1.11 maxprocs=32
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.993159Z lvl=info msg="Using data dir" log_id=0KWwqBfl000 service=store path=/var/lib/influxdb/data
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.993214Z lvl=info msg="Compaction settings" log_id=0KWwqBfl000 service=store max_concurrent_compactions=16 throughput_bytes_per_second=50331648 throughput_bytes_per_second_burst=50331648
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.993230Z lvl=info msg="Open store (start)" log_id=0KWwqBfl000 service=store trace_id=0KWwqC4G000 op_name=tsdb_open op_event=start
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:46.995512Z lvl=info msg="Reading file" log_id=0KWwqBfl000 engine=tsm1 service=cacheloader path=/var/lib/influxdb/wal/_internal/monitor/1/_00001.wal size=84424
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257123Z lvl=info msg="Open store (end)" log_id=0KWwqBfl000 service=store trace_id=0KWwqC4G000 op_name=tsdb_open op_event=end op_elapsed=263.894ms
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257397Z lvl=info msg="Opened service" log_id=0KWwqBfl000 service=subscriber
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257432Z lvl=info msg="Starting monitor service" log_id=0KWwqBfl000 service=monitor
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257453Z lvl=info msg="Registered diagnostics client" log_id=0KWwqBfl000 service=monitor name=build
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257045Z lvl=info msg="Opened shard" log_id=0KWwqBfl000 service=store trace_id=0KWwqC4G000 op_name=tsdb_open index_version=inmem path=/var/lib/influxdb/data/_internal/monitor/1 duration=262.240ms
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257468Z lvl=info msg="Registered diagnostics client" log_id=0KWwqBfl000 service=monitor name=runtime
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257589Z lvl=info msg="Registered diagnostics client" log_id=0KWwqBfl000 service=monitor name=network
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257652Z lvl=info msg="Registered diagnostics client" log_id=0KWwqBfl000 service=monitor name=system
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257728Z lvl=info msg="Starting precreation service" log_id=0KWwqBfl000 service=shard-precreation check_interval=10m advance_period=30m
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257762Z lvl=info msg="Starting snapshot service" log_id=0KWwqBfl000 service=snapshot
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257782Z lvl=info msg="Starting continuous query service" log_id=0KWwqBfl000 service=continuous_querier
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257820Z lvl=info msg="Starting HTTP service" log_id=0KWwqBfl000 service=httpd authentication=false
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.257846Z lvl=info msg="opened HTTP access log" log_id=0KWwqBfl000 service=httpd path=stderr
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.258902Z lvl=info msg="Storing statistics" log_id=0KWwqBfl000 service=monitor db_instance=_internal db_rp=monitor interval=10s
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.258939Z lvl=info msg="Listening on HTTP" log_id=0KWwqBfl000 service=httpd addr=10.0.5.16:8086 https=false
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.259074Z lvl=info msg="Starting retention policy enforcement service" log_id=0KWwqBfl000 service=retention check_interval=30m
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.259565Z lvl=info msg="Listening for signals" log_id=0KWwqBfl000
iml_influxdb.1.rp1o1mg0gk5f@onyx-25.onyx.whamcloud.com    | ts=2020-01-23T18:52:47.260584Z lvl=info msg="Sending usage statistics to usage.influxdata.com" log_id=0KWwqBfl000
```

Notice that the message is not listed in the log output after deploying the stack the second time. This shows that the script is only run during initialization.

Reply:
huh, ok. How do they determine what initial startup is?

Reply:
Not really sure how they do it but the documentation states:

> Warning: scripts in /docker-entrypoint-initdb.d are only run if you start the container with a data > directory that is empty; any pre-existing database will be left untouched on container startup.
- https://hub.docker.com/_/postgres

Reply:
Ok, that makes sense